### PR TITLE
*: reduce the store clone

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve? 
When there are lots of stores, `GetStores` will cost a lot of time due to `Clone`.

### What is changed and how it works?
This PR tries to reduce the store clone when using `GetStore` and `GetStores`.
 
### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: server/core/store.go [hunk 1]
```diff
@@ -294,17 +444,21 @@ func (s *StoreInfo) CompareLocation(other *StoreInfo, labels []string) int {
...
-		for _, label := range s.Labels {
+		for _, label := range storeLabels {
+			fmt.Println("newLabel label", newLabel, label, strings.EqualFold(label.Key, newLabel.Key))
 			if strings.EqualFold(label.Key, newLabel.Key) {
 				label.Value = newLabel.Value
 				continue L
 			}
 		}
-		s.Labels = append(s.Labels, newLabel)
+		fmt.Println("append", newLabel)
```
Comment (disksing) [inline server/core/store.go]: Remove?

File: server/core/store.go [hunk 2]
```diff
@@ -294,17 +444,21 @@ func (s *StoreInfo) CompareLocation(other *StoreInfo, labels []string) int {
...
 // MergeLabels merges the passed in labels with origins, overriding duplicated
 // ones.
-func (s *StoreInfo) MergeLabels(labels []*metapb.StoreLabel) {
+func (s *StoreInfo) MergeLabels(labels []*metapb.StoreLabel) []*metapb.StoreLabel {
+	storeLabels := s.GetLabels()
 L:
 	for _, newLabel := range labels {
-		for _, label := range s.Labels {
+		for _, label := range storeLabels {
+			fmt.Println("newLabel label", newLabel, label, strings.EqualFold(label.Key, newLabel.Key))
```
Comment (disksing) [inline server/core/store.go]: Remove?

File: server/core/store.go (hunk 1 above)
Comment (rleungx) [reply server/core/store.go]: my bad.

**Comment by rleungx** [comment]
Here are the benchmark results:
```
before:
PASS: cluster_test.go:597: testGetStoresSuite.BenchmarkGetStores	    5000	    695042 ns/op
after:
PASS: cluster_test.go:597: testGetStoresSuite.BenchmarkGetStores	  500000	      4624 ns/op
```
PTAL @disksing @nolouch @siddontang

**Comment by huachaohuang** [comment]
The result looks awesome, but please fix the CI.

**Comment by rleungx** [comment]
The CI problem is not related to this PR. @huachaohuang

**Pull request merged**